sql/tests: internal error in row-by-row windower when TSQuery is in PARTITION BY

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/14127427?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/14127427?buildTab=artifacts#/) on master @ [d4306a9e895eaf50ef9b87b4ba09581447f7dd70](https://github.com/cockroachdb/cockroach/commits/d4306a9e895eaf50ef9b87b4ba09581447f7dd70):

Random syntax error:

```
    rsg_test.go:909: Crash detected: server panic: pq: internal error: unable to encode table key: *tree.DTSQuery
```
Query:

```
        WITH
        	"w'ith_615" (col_3319)
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						('''UsRrdunxc'' & ''HBjpjkTM'' <-> ''KI''':::TSQUERY),
        						('''bkChMACS''':::TSQUERY),
        						('''QLuuhzwS'' <-> ''ZWfUiVcUmP'' & ''bHZ'' & ''EtqKlNkbp'' <-> ''DNJmJeo''':::TSQUERY)
        				)
        					AS tab_1525 (col_3319)
        		)
        SELECT
        	"tab\\x5d_1526"._date AS col_3320,
        	jsonb_object_agg('{"c": true, "faDqzlHb": [false], "qJPcvwFod": false}':::JSONB::STRING, "t'ab_1527"._jsonb)::JSONB
        		AS """col_3321",
        	merge_statement_stats(tab_1528._jsonb::JSONB) OVER (PARTITION BY tab_1528._jsonb, 😞cte_rḙf_183.col_3319, "tab\\x5d_1526"._int8 ORDER BY tab_1528._jsonb ASC)::JSONB
        		AS co̡l_3322,
        	xor_agg("tab\\x5d_1526"._int8::INT8)::INT8 AS "cOl_3323",
        	'!''zrEevzFlh'' <-> !''sZw'' & !''P'' | ''vqw''':::TSQUERY AS col_3324
        FROM
        	defaultdb.public.seed@[0] AS "tab\\x5d_1526",
        	"w'ith_615" AS 😞cte_rḙf_183,
        	defaultdb.public.seed AS "t'ab_1527"
        	JOIN defaultdb.public.seed AS tab_1528 ON ("t'ab_1527"._jsonb) = (tab_1528._jsonb)
        WHERE
        	st_ispolygonccw('010700000009000000010500000001000000010200000003000000AD3167F4906AFFC140BEF705BEABBEC1A0F1AC478CF0D541789AD7E6B7A8DE414A3DE968566BF4414AB28DD5AB96F8410101000000DE120D2B8BC7F241165957F60DF9F9C101050000000100000001020000000600000056A1B30B512E00422C5E3A6AB8C7FDC1324018B9CE2B0242E0316B7AF6F3BEC144E0A26FDF9DFB41B0BE48FD7F44FD41188E41A210DFE74184742EB0B9EEEC41F0EF3E1443CEC44148E2ED4420A7DF413F32E53EDD96F1C11A4FA08EE4BFF24101040000000500000001010000002CAE239C282FFB415462F9F32D9700C20101000000CA85BFD7496CE4C18A84EE7CC93201420101000000362A606C5B87004228C2AE5B1C9DDB4101010000001041FFD58A3FEE41A57FFDDA064301C20101000000C4BBD0075716EC4112CA6F4BBA26F1410103000000010000000600000008FF2B3E7371E1414D6C5F56CAB001C2AAF2FAA52385F0412A449DF2BB65F241F401E0F8BE6BF44136E25A23775C0042B088E0C448DEC541B0B06F211488D94122E9BA183E3CE0C1102D5F36B8D5D84108FF2B3E7371E1414D6C5F56CAB001C201050000000200000001020000000200000010DD71F5A738FE4150938A06555CCF410C2FD545165D0242D0555E5AC34BEA41010200000003000000C077C452E96ADDC19C07A5E7B5D4FCC1352D8A21A2DDF4C146229EE5F658F941F7A9C27AA94402C2ACED51843590EC41010200000004000000FC82735FC99FF241273CB673DBF9FDC1F8517155715EFC41B83CC8278239FDC14EB8A83E84DBF8417E6A3060300CEDC146AF892652A4F84186783102D616E7C101010000005C03AE1E03C2EBC1949027BEBD15E0410101000000E869B9166DD0E741AF5688C9215402C2':::GEOMETRY::GEOMETRY)::BOOL
        GROUP BY
        	"tab\\x5d_1526"._date, 😞cte_rḙf_183.col_3319, "t'ab_1527"._jsonb, tab_1528._jsonb, "tab\\x5d_1526"._int8
        LIMIT
        	85:::INT8;
```
Schema:

```
    rsg_test.go:718: To reproduce, use schema:
    rsg_test.go:720: SET sql_safe_updates = false;;
    rsg_test.go:720: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:720: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:720: CREATE TABLE table_1 ("co�l1_0" FLOAT4, "co	l1_1" TIMESTAMPTZ NOT NULL, col1_2 TSQUERY NULL, col😧1_3 FLOAT8 NULL, "cOl1_4" DECIMAL NOT NULL, ".c😲ol1_5" FLOAT8 AS ("co�l1_0" + col😧1_3) VIRTUAL, "c\\U00005B52o😙l1_6" STRING NOT NULL AS (CASE WHEN "co	l1_1" IS NULL THEN '!n':::STRING ELSE e'1\x1b)yBg]t\x0b':::STRING END) STORED, " \\U000FBE95col1 	_7" FLOAT8 AS ("co�l1_0" + col😧1_3) STORED, INDEX (col😧1_3 DESC) WHERE ((((((table_1.col😧1_3 >= 5e-324:::FLOAT8) AND (table_1." \\U000FBE95col1 	_7" < '-Inf':::FLOAT8)) OR (table_1.".c😲ol1_5" > 1.401298464324817e-45:::FLOAT8)) OR (table_1."c\\U00005B52o😙l1_6" = 'X':::STRING)) AND (table_1."cOl1_4" >= (-1.0):::DECIMAL)) OR (table_1."co�l1_0" != 1.401298464324817e-45:::FLOAT8)) AND (table_1."co	l1_1" >= '0001-01-01 00:00:00+00':::TIMESTAMPTZ), INDEX ("c\\U00005B52o😙l1_6" ASC, " \\U000FBE95col1 	_7"), UNIQUE ("cOl1_4" DESC) STORING ("co�l1_0", "co	l1_1", col1_2, col😧1_3, "c\\U00005B52o😙l1_6", " \\U000FBE95col1 	_7") PARTITION BY LIST ("cOl1_4") (PARTITION "}table_1_part_0" VALUES IN (((-2.29113848883650696E+35):::DECIMAL,), (85243401280628.33227:::DECIMAL,), ((-41660726.88097203502):::DECIMAL,), (11093014952822807.24:::DECIMAL,), ((-2001663687232045.947):::DECIMAL,), ((-9.58252336621602489E+25):::DECIMAL,), ((-1):::DECIMAL,)), PARTITION table_1_part_1 VALUES IN (((-7226.760485856768436):::DECIMAL,), ((-2.035161646871643564):::DECIMAL,), (1737520690.154457798:::DECIMAL,), (700.7855668992795774:::DECIMAL,), ((-8457164025182115811):::DECIMAL,), ((-8.48394907347344243E+19):::DECIMAL,), ((-6780304717.011586697):::DECIMAL,)), PARTITION table_1_part_2 VALUES IN ((5.323948193068594733E+19:::DECIMAL,), (1.665644755286440861E+28:::DECIMAL,), ((-555594261100821923.4):::DECIMAL,), (7.838636997565398707E+20:::DECIMAL,), ((-54490.318460499286):::DECIMAL,), (3.466178886086274062E+30:::DECIMAL,), ((-7.231917608501195126E+28):::DECIMAL,)), PARTITION "table_1_p art_3" VALUES IN ((4.1949595627487489E+30:::DECIMAL,), ((-7307163730918927.719):::DECIMAL,), (225818.2451286123653:::DECIMAL,), (66117894370541755.05:::DECIMAL,), (69813033770179497.5:::DECIMAL,), (3.445159096712812487E+28:::DECIMAL,), ((-13.63206150463538144):::DECIMAL,)), PARTITION table_1_part_4 VALUES IN ((1.0:::DECIMAL,), (0:::DECIMAL,), (7368320359779477.69:::DECIMAL,), (6.930610873773024661E+20:::DECIMAL,), (6079.28261494412093:::DECIMAL,), (615573358136.7989845:::DECIMAL,), ((-4.460849026273673830E+26):::DECIMAL,)), PARTITION table_1_part_5 VALUES IN ((2380.099321845564078:::DECIMAL,), (8.603379137526811104E+29:::DECIMAL,), ((-7.773078328344578224E+27):::DECIMAL,), (3.745894852066021165E+31:::DECIMAL,), (7830.76610822354787:::DECIMAL,), (2.063693807603453442E+26:::DECIMAL,))), UNIQUE ((col😧1_3 + "co�l1_0") DESC, ".c😲ol1_5" ASC, (CASE WHEN "co	l1_1" IS NULL THEN '6':::STRING ELSE '=~':::STRING END) DESC) STORING (col😧1_3, " \\U000FBE95col1 	_7"), UNIQUE ("cOl1_4" ASC, col😧1_3 ASC) STORING ("c\\U00005B52o😙l1_6", " \\U000FBE95col1 	_7"), UNIQUE (".c😲ol1_5" ASC), INDEX ("co	l1_1" ASC, "cOl1_4" ASC) WHERE (((table_1."c\\U00005B52o😙l1_6" = '':::STRING) OR (table_1.".c😲ol1_5" < 1.401298464324817e-45:::FLOAT8)) OR (table_1."co�l1_0" != 1.0:::FLOAT8)) OR (table_1."co	l1_1" <= '3000-01-01 00:00:00+00':::TIMESTAMPTZ), FAMILY ("cOl1_4"), FAMILY ("co�l1_0", " \\U000FBE95col1 	_7"), FAMILY ("co	l1_1"), FAMILY (col😧1_3, col1_2), FAMILY ("c\\U00005B52o😙l1_6"));
    rsg_test.go:720: CREATE TABLE table😗_2 ("Col2_0" FLOAT8, col2_1 TIMETZ NOT NULL, "col(2_2" TIMESTAMPTZ NOT NULL, col2_3 TIMETZ NULL, "col2""_4" INT2 NOT NULL, col2_5 REFCURSOR NULL, "'col2_6" STRING NOT NULL AS (lower(CAST(col2_5 AS STRING))) VIRTUAL, PRIMARY KEY ("'col2_6" ASC, "col(2_2" ASC), UNIQUE ("col2""_4" DESC) STORING ("Col2_0", col2_1) PARTITION BY LIST ("col2""_4") (PARTITION "t a😍ble😗_2_part_0" VALUES IN ((31509:::INT8,), ((-18851):::INT8,), (30827:::INT8,), (19071:::INT8,), ((-23783):::INT8,), (12443:::INT8,), (28383:::INT8,), (31301:::INT8,), ((-10016):::INT8,)), PARTITION table😗_2_part_1 VALUES IN (((-31514):::INT8,), ((-8225):::INT8,), ((-12010):::INT8,), ((-20074):::INT8,), (18234:::INT8,), ((-28380):::INT8,), ((-31870):::INT8,), (3521:::INT8,), ((-25007):::INT8,)), PARTITION "table😗_2\\uF452_part_2" VALUES IN ((10934:::INT8,), ((-2537):::INT8,), (11234:::INT8,), (30757:::INT8,), ((-19619):::INT8,), ((-4306):::INT8,), (17285:::INT8,), (23859:::INT8,), (0:::INT8,)), PARTITION table😗_2_part_3 VALUES IN (((-809):::INT8,), ((-24056):::INT8,), ((-26843):::INT8,), ((-25102):::INT8,), (16092:::INT8,), (18858:::INT8,), (14225:::INT8,), (13655:::INT8,)), PARTITION "/table😗_2
_part_4" VALUES IN ((62:::INT8,), (13673:::INT8,), (21409:::INT8,), ((-17364):::INT8,), (19665:::INT8,), (4944:::INT8,), ((-18092):::INT8,), (17830:::INT8,)), PARTITION table😗_2_part_5 VALUES IN (((-19203):::INT8,), ((-19756):::INT8,), (24602:::INT8,), ((-5308):::INT8,), (24831:::INT8,), (6334:::INT8,), (32383:::INT8,), ((-4888):::INT8,)), PARTITION "DEFAULT" VALUES IN ((DEFAULT,))) WHERE table😗_2."col2""_4" = (-1):::INT8, UNIQUE ("col2""_4") STORING (col2_3) WHERE (((((table😗_2.col2_1 != '00:00:00+15:59':::TIMETZ) AND (table😗_2."col2""_4" > 1:::INT8)) AND (table😗_2."'col2_6" < 'X':::STRING)) AND (table😗_2.col2_3 = '24:00:00-15:59':::TIMETZ)) OR (table😗_2."col(2_2" > '294276-12-31 23:59:59.999999+00':::TIMESTAMPTZ)) AND (table😗_2."Col2_0" != 5e-324:::FLOAT8), INDEX ("col2""_4" DESC, "col(2_2" DESC) STORING ("Col2_0", col2_5) WHERE (table😗_2."Col2_0" > 1.401298464324817e-45:::FLOAT8) OR (table😗_2.col2_3 = '24:00:00-15:59':::TIMETZ), UNIQUE ("col2""_4" ASC) STORING (col2_5));
    rsg_test.go:720: CREATE TABLE table_3 (col3_0 TIMESTAMPTZ NULL, col3_1 INT8 NOT NULL, col3_2 VARCHAR NOT NULL, "col\\U0000DA813_3" STRING COLLATE de_DE NOT NULL, "\rcol3_4" FLOAT4 NOT NULL, col3_5 BOOL, "col3%09_6" TSVECTOR NOT NULL, col3_7 PG_LSN NOT NULL, col3_8 INT2 NOT NULL, "co\\x29l3_9" GEOGRAPHY NOT NULL, "coL3_10" INT8 NOT NULL AS (col3_1 + 3998989378353666387:::INT8) STORED, "c😱ol%q3_11" STRING NOT NULL AS (lower(CAST(col3_7 AS STRING))) STORED, col3_12 INT8 NOT NULL AS (col3_1 + col3_8) STORED, INDEX (col3_2, col3_8 ASC), UNIQUE (col3_2 DESC, col3_0 ASC) WHERE (((((table_3.col3_0 < '3000-01-01 00:00:00+00':::TIMESTAMPTZ) OR (table_3."\rcol3_4" <= '+Inf':::FLOAT8)) OR (table_3."coL3_10" <= 0:::INT8)) OR (table_3."c😱ol%q3_11" > '':::STRING)) OR (NOT table_3.col3_5)) AND (table_3.col3_1 <= 1:::INT8), FAMILY (col3_12, col3_0, col3_5), FAMILY (col3_1), FAMILY (col3_8), FAMILY (col3_7, "\rcol3_4"), FAMILY ("c😱ol%q3_11"), FAMILY ("col3%09_6", "co\\x29l3_9"), FAMILY ("coL3_10", col3_2), FAMILY ("col\\U0000DA813_3"));
    rsg_test.go:720: CREATE TABLE tabͨle_4 ("col\n4_0" INT4, "%vcol""4_1" STRING NOT NULL, col4_2 REGROLE NULL, "col4�_3" PG_LSN NOT NULL, "col4%q_4" GEOMETRY, col4_5 GEOMETRY NOT NULL, col4_6 INT8 NOT NULL, col4_7 GEOGRAPHY NOT NULL, col4_8 INT8 NOT NULL, "Col 4͕_9" INT8 AS (("col\n4_0" + col4_8) + col4_6) VIRTUAL, col4_10 INT8 AS ("col\n4_0" + col4_6) STORED, PRIMARY KEY (col4_8 DESC), INVERTED INDEX ("Col 4͕_9" DESC, "%vcol""4_1" DESC, col4_7 ASC), INVERTED INDEX ("col\n4_0" ASC, col4_8 ASC, col4_5) WHERE (((((tabͨle_4."Col 4͕_9" > 1:::INT8) AND (tabͨle_4.col4_8 >= (-32768):::INT8)) AND (tabͨle_4.col4_10 > (-9223372036854775807):::INT8)) OR (tabͨle_4.col4_6 < (-9223372036854775807):::INT8)) OR (tabͨle_4."%vcol""4_1" >= 'X':::STRING)) AND (tabͨle_4."col\n4_0" < (-1):::INT8), INDEX ("col4�_3" ASC) STORING ("%vcol""4_1", col4_2, "col4%q_4", col4_5, col4_6, col4_10) WHERE (((((tabͨle_4."Col 4͕_9" >= 127:::INT8) OR (tabͨle_4."%vcol""4_1" = e'\U00002603':::STRING)) AND (tabͨle_4.col4_10 != 1:::INT8)) AND (tabͨle_4.col4_8 <= 2147483647:::INT8)) AND (tabͨle_4."col\n4_0" < 32767:::INT8)) OR (tabͨle_4.col4_6 = (-1):::INT8), INVERTED INDEX (col4_5 ASC) WHERE (((((tabͨle_4.col4_8 > (-9223372036854775807):::INT8) AND (tabͨle_4.col4_10 < 32767:::INT8)) AND (tabͨle_4."col\n4_0" >= (-1):::INT8)) AND (tabͨle_4."Col 4͕_9" != (-9223372036854775807):::INT8)) AND (tabͨle_4.col4_6 != (-9223372036854775807):::INT8)) OR (tabͨle_4."%vcol""4_1" < '"':::STRING) NOT VISIBLE, INDEX ("col4�_3", col4_2, col4_10 ASC) WHERE (tabͨle_4."Col 4͕_9" > (-9223372036854775807):::INT8) AND (tabͨle_4.col4_6 = (-9223372036854775807):::INT8), INVERTED INDEX (col4_7) VISIBILITY 0.22, INVERTED INDEX (col4_5 ASC), FAMILY (col4_7, "col4%q_4"), FAMILY (col4_10), FAMILY ("col4�_3", col4_2), FAMILY (col4_5), FAMILY (col4_6, "col\n4_0"), FAMILY (col4_8), FAMILY ("%vcol""4_1"));
    rsg_test.go:720: CREATE TABLE "tab
le_5" ("%pc%pol	5_0" NAME NOT NULL, "c""ol5_1" STRING NOT NULL, "col 5%v_2" STRING NOT NULL AS (lower("c""ol5_1")) STORED, PRIMARY KEY ("c""ol5_1" ASC), INDEX ("%pc%pol	5_0", "c""ol5_1" ASC) STORING ("col 5%v_2") NOT VISIBLE, INDEX ("col 5%v_2" ASC), INVERTED INDEX ("%pc%pol	5_0" gin_trgm_ops ASC) WHERE "tab
le_5"."col 5%v_2" <= e'\U00002603':::STRING, INDEX ("col 5%v_2" DESC, lower("%pc%pol	5_0")) WHERE "tab
le_5"."c""ol5_1" >= '':::STRING, UNIQUE ("col 5%v_2", lower("c""ol5_1") ASC, lower("%pc%pol	5_0")), UNIQUE ("%pc%pol	5_0" ASC) STORING ("col 5%v_2"), INDEX (lower("%pc%pol	5_0") ASC, lower("c""ol5_1")) WHERE (("tab
le_5"."c""ol5_1" > '"':::STRING) AND ("tab
le_5"."col 5%v_2" > '"':::STRING)) OR ("tab
le_5"."%pc%pol	5_0" < e'\'':::STRING:::NAME), FAMILY ("c""ol5_1"), FAMILY ("col 5%v_2", "%pc%pol	5_0"));
    rsg_test.go:720: ALTER TABLE table_1 INJECT STATISTICS e'[{"avg_size": 11, "columns": ["\\" \\\\\\\\U000FBE95col1 \\t_7\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1365447845652085999, "histo_col_type": "", "name": "__auto__", "null_count": 2609091025649164555, "row_count": 4119016737157927986}, {"avg_size": 26, "columns": ["\\"co\\u000cl1_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1760934822040606135, "histo_col_type": "", "name": "__auto__", "null_count": 208576967693066272, "row_count": 4119016737157927986}, {"avg_size": 4, "columns": ["\\"co\\tl1_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3932061143237061010, "histo_buckets": [{"distinct_range": 0, "num_eq": 500000, "num_range": 0, "upper_bound": "2001-10-12 09:47:44.000502+00"}, {"distinct_range": 18.279373063034143, "num_eq": 3790777735764410385, "num_range": 80, "upper_bound": "2006-06-24 05:28:00.000311+00"}, {"distinct_range": 0, "num_eq": 2427879125427617279, "num_range": 3253990496901985728, "upper_bound": "2024-11-09 22:54:40.000453+00"}, {"distinct_range": 0, "num_eq": 2080498285768228333, "num_range": 8192518836249372285, "upper_bound": "2021-09-25 02:35:33.000931+00"}], "histo_col_type": "TIMESTAMPTZ", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 4119016737157927986}, {"avg_size": 19, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3181750590513791230, "histo_col_type": "", "name": "__auto__", "null_count": 3546295619555188181, "row_count": 4119016737157927986}, {"avg_size": 31, "columns": ["col\U0001F6271_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1068191064867921606, "histo_col_type": "", "name": "__auto__", "null_count": 2884746807253691681, "row_count": 4119016737157927986}, {"avg_size": 1, "columns": ["\\"cOl1_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2008102490030776633, "histo_buckets": [{"distinct_range": 0, "num_eq": 6525264803522933019, "num_range": 0, "upper_bound": "-1.234E+401"}, {"distinct_range": 0, "num_eq": 7999376701984488749, "num_range": 40000000000, "upper_bound": "-229741715371134.2145"}, {"distinct_range": 0, "num_eq": 6000000000, "num_range": 30000, "upper_bound": "-0.03765341123534638817"}, {"distinct_range": 0, "num_eq": 5262682630645052149, "num_range": 2297226993307504174, "upper_bound": "58676688.34851710328"}, {"distinct_range": 2736107637703792000, "num_eq": 8056777279270798776, "num_range": 7083688851654538093, "upper_bound": "3084784291.010401749"}, {"distinct_range": 615.0926593636678, "num_eq": 5078431683550471327, "num_range": 20000, "upper_bound": "503310373476963481"}, {"distinct_range": 0, "num_eq": 2652348200360302819, "num_range": 1000000000, "upper_bound": "4161162559532306632"}], "histo_col_type": "DECIMAL", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 4119016737157927986}, {"avg_size": 10, "columns": ["\\".c\U0001F632ol1_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2362197093852168413, "histo_buckets": [{"distinct_range": 0, "num_eq": 40000000000, "num_range": 0, "upper_bound": "0.4974566781589896"}, {"distinct_range": 1383885191902843000, "num_eq": 9009150704700681177, "num_range": 1383885191902842907, "upper_bound": "0.5851533846840632"}, {"distinct_range": 1222482.7008229194, "num_eq": 70000000, "num_range": 2000000, "upper_bound": "1.0"}, {"distinct_range": 8000, "num_eq": 5722459565872234736, "num_range": 8000, "upper_bound": "1.352153196998159"}, {"distinct_range": 0, "num_eq": 6908148642551264735, "num_range": 6745958286263287057, "upper_bound": "1.418129392692878"}, {"distinct_range": 7690368676250147000, "num_eq": 500000, "num_range": 7690368676250146799, "upper_bound": "-0.2789158532597844"}, {"distinct_range": 5566911551.35069, "num_eq": 7592716003728492209, "num_range": 7000000000, "upper_bound": "-0.3193595328790094"}, {"distinct_range": 0, "num_eq": 9000000000, "num_range": 0, "upper_bound": "-0.37258147972755423"}, {"distinct_range": 22045.473389681905, "num_eq": 20000000, "num_range": 80000, "upper_bound": "-0.776903699627618"}], "histo_col_type": "FLOAT8", "histo_version": 3, "name": "__auto__", "null_count": 1603821648530213206, "row_count": 4119016737157927986}, {"avg_size": 11, "columns": ["\\"c\\\\\\\\U00005B52o\U0001F619l1_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 839105860531720866, "histo_buckets": [{"distinct_range": 0, "num_eq": 7643942057577256089, "num_range": 0, "upper_bound": "\\u000e"}, {"distinct_range": 0, "num_eq": 50, "num_range": 3188735023907420765, "upper_bound": "\\u001b"}, {"distinct_range": 0, "num_eq": 1000000, "num_range": 0, "upper_bound": "^EA"}, {"distinct_range": 0, "num_eq": 1567765772197515846, "num_range": 6407285749365903414, "upper_bound": "NX\\u000ex"}, {"distinct_range": 0, "num_eq": 1000000000, "num_range": 3085686485374357490, "upper_bound": "\'>U\\u000b."}, {"distinct_range": 4730235924504145000, "num_eq": 100000, "num_range": 6080783236948421706, "upper_bound": ";[\\\\\\u000b*,\\u001e"}, {"distinct_range": 4957075653699045000, "num_eq": 9000, "num_range": 4957075653699045716, "upper_bound": "\\u0001P$u\\u0016``\\"|"}, {"distinct_range": 0, "num_eq": 7432163254996394617, "num_range": 0, "upper_bound": "z\\u0001r\\u0012/F?]"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 4119016737157927986}]':::JSONB;
    rsg_test.go:720: ALTER TABLE table😗_2 INJECT STATISTICS e'[{"avg_size": 18, "columns": ["\\"col2\\"\\"_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 99, "histo_buckets": [{"distinct_range": 0, "num_eq": 9000000, "num_range": 0, "upper_bound": "0"}, {"distinct_range": 3784266098706867700, "num_eq": 5397310483271485108, "num_range": 3784266098706867611, "upper_bound": "10314"}, {"distinct_range": 0, "num_eq": 5042879150589816239, "num_range": 928043303170895618, "upper_bound": "-21963"}, {"distinct_range": 30000, "num_eq": 1000000000, "num_range": 30000, "upper_bound": "21451"}, {"distinct_range": 0, "num_eq": 700000, "num_range": 50000000000, "upper_bound": "11795"}, {"distinct_range": 32430.971661770564, "num_eq": 400, "num_range": 6000000, "upper_bound": "3892"}, {"distinct_range": 0, "num_eq": 800000, "num_range": 9000000, "upper_bound": "-15801"}, {"distinct_range": 1207035415012715, "num_eq": 30000000, "num_range": 1207035415012715, "upper_bound": "32767"}], "histo_col_type": "INT2", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 200}, {"avg_size": 10, "columns": ["col2_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 48, "histo_col_type": "", "name": "__auto__", "null_count": 106, "row_count": 200}, {"avg_size": 15, "columns": ["\\"\'col2_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 51, "histo_buckets": [{"distinct_range": 0, "num_eq": 700000000, "num_range": 0, "upper_bound": "\\u001bA"}, {"distinct_range": 0, "num_eq": 8683842149412346013, "num_range": 10, "upper_bound": ";\\u000elrY"}, {"distinct_range": 2054721305167746600, "num_eq": 3865194385407639513, "num_range": 8940971048202867416, "upper_bound": "Nt})\\u000eaZ"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 200}, {"avg_size": 29, "columns": ["\\"Col2_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 31, "histo_col_type": "", "name": "__auto__", "null_count": 182, "row_count": 200}, {"avg_size": 30, "columns": ["col2_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 85, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 200}, {"avg_size": 23, "columns": ["\\"col(2_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 27, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 200}, {"avg_size": 27, "columns": ["col2_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 23, "histo_col_type": "", "name": "__auto__", "null_count": 49, "row_count": 200}]':::JSONB;
    rsg_test.go:720: ALTER TABLE table_3 INJECT STATISTICS e'[{"avg_size": 2, "columns": ["\\"coL3_10\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3088, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 1, "columns": ["\\"col\\\\\\\\U0000DA813_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 206, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 9, "columns": ["\\"\\\\rcol3_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 689, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 22, "columns": ["\\"co\\\\\\\\x29l3_9\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1412, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 18, "columns": ["\\"col3%09_6\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4838, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 23, "columns": ["col3_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1256, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 28, "columns": ["col3_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1087, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 9, "columns": ["col3_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2433, "histo_buckets": [{"distinct_range": 0, "num_eq": 2280152348460112600, "num_range": 0, "upper_bound": ""}, {"distinct_range": 200, "num_eq": 4822205684453697270, "num_range": 200, "upper_bound": "GQ\\u0006"}, {"distinct_range": 0, "num_eq": 100000000000, "num_range": 8487522009780618024, "upper_bound": "\\u000e2~\\u001b"}, {"distinct_range": 0, "num_eq": 9019677463857572618, "num_range": 40000000000, "upper_bound": "\\u0008\\u0007]i\\u001b"}, {"distinct_range": 2803032212255622700, "num_eq": 100, "num_range": 5716572388289238097, "upper_bound": "L_:9\\n"}, {"distinct_range": 0, "num_eq": 60000, "num_range": 6034127864799263648, "upper_bound": "h\\u000c\\u00037x"}, {"distinct_range": 0, "num_eq": 3000000000, "num_range": 2986235616455378040, "upper_bound": "/>\\u0008!,>J"}, {"distinct_range": 0, "num_eq": 7336311442122793692, "num_range": 0, "upper_bound": "h#Rlhx==\\u0017"}], "histo_col_type": "VARCHAR", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 10, "columns": ["col3_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3122, "histo_col_type": "", "name": "__auto__", "null_count": 3127, "row_count": 5000}, {"avg_size": 15, "columns": ["col3_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1759, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 11, "columns": ["col3_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4425, "histo_col_type": "", "name": "__auto__", "null_count": 1397, "row_count": 5000}, {"avg_size": 14, "columns": ["col3_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3661, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}, {"avg_size": 15, "columns": ["\\"c\U0001F631ol%q3_11\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2922, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 5000}]':::JSONB;
    rsg_test.go:720: ALTER TABLE tabͨle_4 INJECT STATISTICS e'[{"avg_size": 27, "columns": ["col4_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 13924, "histo_buckets": [{"distinct_range": 0, "num_eq": 40, "num_range": 0, "upper_bound": "\\\\x42fdffffffffffffffff01c1f3fadff2731391c1f341d443d444ef"}, {"distinct_range": 331021716237668100, "num_eq": 3731152849146009919, "num_range": 1218615023922743296, "upper_bound": "\\\\x42fdffffffffffffffff01c2003769e399df9e41fb198d723f2bb8"}, {"distinct_range": 2232580115419121700, "num_eq": 90000, "num_range": 5231301811902364837, "upper_bound": "\\\\x42fd040000000000000000c1f737ea2d08bae8c1fa4830434de5394201c0ba5fca01144200f2ab6f076786"}, {"distinct_range": 1268031481287780000, "num_eq": 9184125448170515881, "num_range": 1268031481287780027, "upper_bound": "\\\\x42fd0c0000000000000000c201c5fbf27122c7c1f9b44b0d0378fe41ff4c7eafce91ac4201e5bb4e407b20"}, {"distinct_range": 3168385163129130000, "num_eq": 1000000, "num_range": 4742326522673183933, "upper_bound": "\\\\x42fd0f0000000000000000c1f737ea2d08bae8c1fa4830434de5394201c0ba5fca01144200f2ab6f076786"}, {"distinct_range": 0, "num_eq": 6776283944370005753, "num_range": 0, "upper_bound": "\\\\x42fd100000000000000000c2025bc42d897162c2028faf9db863aa4201c19627dd86b24201f2e9cbef62d0"}, {"distinct_range": 0, "num_eq": 5067637463357509348, "num_range": 1000000, "upper_bound": "\\\\x42fd12c000000000000000c201c5fbf27122c7c1f9b44b0d0378fe41ff4c7eafce91ac4201e5bb4e407b20"}, {"distinct_range": 535681871744367600, "num_eq": 3441645242368317517, "num_range": 574732363306627730, "upper_bound": "\\\\x42fd134000000000000000c201c5fbf27122c7c1f9b44b0d0378fe41ff4c7eafce91ac4201e5bb4e407b20"}, {"distinct_range": 0, "num_eq": 900, "num_range": 6759116738294168592, "upper_bound": "\\\\x42fd140000000000000000c1f737ea2d08bae8c1fa4830434de5394201c0ba5fca01144200f2ab6f076786"}, {"distinct_range": 10774238.056079928, "num_eq": 70000000000, "num_range": 40000000, "upper_bound": "\\\\x42fd14ac00000000000000c201c5fbf27122c7c1f9b44b0d0378fe41ff4c7eafce91ac4201e5bb4e407b20"}, {"distinct_range": 2002964704116364500, "num_eq": 0, "num_range": 4173846887663258347, "upper_bound": "\\\\x42fd150ad0000000000000c20141f8689c8c77c20203f7f02c1a27420247a44ed99e8641feabf79875cfc4"}, {"distinct_range": 700, "num_eq": 3977341056870334649, "num_range": 700, "upper_bound": "\\\\x42fd157500000000000000c20141f8689c8c77c20203f7f02c1a27420247a44ed99e8641feabf79875cfc4"}, {"distinct_range": 84934983.18307719, "num_eq": 2765646612791904919, "num_range": 100000000, "upper_bound": "\\\\x42fd158400000000000000c20141f8689c8c77c20203f7f02c1a27420247a44ed99e8641feabf79875cfc4"}, {"distinct_range": 7588704576660404000, "num_eq": 300, "num_range": 7588704576660404676, "upper_bound": "\\\\x42fd159c00000000000000c20141f8689c8c77c20203f7f02c1a27420247a44ed99e8641feabf79875cfc4"}, {"distinct_range": 4070287115357917700, "num_eq": 7000000, "num_range": 4070287115357917487, "upper_bound": "\\\\x42fd18900000000000000041d963262603b568c1d12c5ea80ee05841ffed81257caeb441fa92fda5a21e18"}, {"distinct_range": 582094540971491800, "num_eq": 7000, "num_range": 582094540971491853, "upper_bound": "\\\\x42fd18b00000000000000041d963262603b568c1d12c5ea80ee05841ffed81257caeb441fa92fda5a21e18"}, {"distinct_range": 3673097.797669675, "num_eq": 5358581876866879376, "num_range": 7000000, "upper_bound": "\\\\x42fd18cc0000000000000041d963262603b568c1d12c5ea80ee05841ffed81257caeb441fa92fda5a21e18"}, {"distinct_range": 20000, "num_eq": 5290601306374731462, "num_range": 20000, "upper_bound": "\\\\x42fd1c0000000000000000c1f737ea2d08bae8c1fa4830434de5394201c0ba5fca01144200f2ab6f076786"}, {"distinct_range": 18.721430364927325, "num_eq": 763999740971377311, "num_range": 20, "upper_bound": "\\\\x42fd1f300000000000000041d963262603b568c1d12c5ea80ee05841ffed81257caeb441fa92fda5a21e18"}, {"distinct_range": 422713079871851650, "num_eq": 1191545745613240888, "num_range": 422713079871851672, "upper_bound": "\\\\x42fdffffffffffffffff0041d963262603b568c1d12c5ea80ee05841ffed81257caeb441fa92fda5a21e18"}, {"distinct_range": 3032517366149814300, "num_eq": 6070053040507985647, "num_range": 3032517366149814111, "upper_bound": "\\\\x42fdffffffffffffffff00c1a1e26c929da88041bb28d43f04698041fd75c3f82fbf0041f2e48dd1562e1c"}, {"distinct_range": 90000, "num_eq": 10000000, "num_range": 90000, "upper_bound": "\\\\x42fdffffffffffffffff00c1f737ea2d08bae8c1fa4830434de5394201c0ba5fca01144200f2ab6f076786"}, {"distinct_range": 100000, "num_eq": 6757842531825116587, "num_range": 100000, "upper_bound": "\\\\x42fdffffffffffffffff00c20141f8689c8c77c20203f7f02c1a27420247a44ed99e8641feabf79875cfc4"}, {"distinct_range": 1041818861912196900, "num_eq": 7015696124539460038, "num_range": 1041818861912196914, "upper_bound": "\\\\x42fdffffffffffffffff00c201c5fbf27122c7c1f9b44b0d0378fe41ff4c7eafce91ac4201e5bb4e407b20"}, {"distinct_range": 0, "num_eq": 2256461322030505296, "num_range": 100, "upper_bound": "\\\\x42fdffffffffffffffff00c2025bc42d897162c2028faf9db863aa4201c19627dd86b24201f2e9cbef62d0"}], "histo_col_type": "BYTES", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 30000}, {"avg_size": 1, "columns": ["col4_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 661, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30000}, {"avg_size": 15, "columns": ["col4_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4258, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30000}, {"avg_size": 25, "columns": ["\\"Col 4\u0355_9\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 9841, "histo_buckets": [{"distinct_range": 0, "num_eq": 8083991283215886912, "num_range": 0, "upper_bound": "1"}, {"distinct_range": 0, "num_eq": 60000000, "num_range": 1640574608235144600, "upper_bound": "-3210186927787952990"}, {"distinct_range": 0, "num_eq": 800000, "num_range": 5993751510951110042, "upper_bound": "-1577166182389481391"}, {"distinct_range": 3605180441200428500, "num_eq": 4678864424812490520, "num_range": 7225114531890788632, "upper_bound": "6140964347663815026"}, {"distinct_range": 3255962650116480500, "num_eq": 3182393945105461758, "num_range": 7464495342195207148, "upper_bound": "-4501026450584994099"}, {"distinct_range": 4655025488129789000, "num_eq": 2821063202140035794, "num_range": 4655025488129789032, "upper_bound": "4644663183100158327"}, {"distinct_range": 25.85557225949888, "num_eq": 591386176303495404, "num_range": 50, "upper_bound": "1556497361425869626"}], "histo_col_type": "INT8", "histo_version": 3, "name": "__auto__", "null_count": 19018, "row_count": 30000}, {"avg_size": 6, "columns": ["\\"col\\\\n4_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 10667, "histo_col_type": "", "name": "__auto__", "null_count": 18454, "row_count": 30000}, {"avg_size": 31, "columns": ["\\"%vcol\\"\\"4_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 27161, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 30000}, {"avg_size": 3, "columns": ["col4_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 13885, "histo_col_type": "", "name": "__auto__", "null_count": 993, "row_count": 30000}, {"avg_size": 16, "columns": ["\\"col4\\u000c_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 8869, "histo_buckets": [{"distinct_range": 0, "num_eq": 1193941629651279925, "num_range": 0, "upper_bound": "FFFFFFFF/FFFFFFFF"}, {"distinct_range": 0, "num_eq": 798796119674170214, "num_range": 60000000000, "upper_bound": "3D484415/B9C0110A"}, {"distinct_range": 0, "num_eq": 4417088573712234592, "num_range": 10000000000, "upper_bound": "ADFCD14B/81E275AF"}, {"distinct_range": 0, "num_eq": 2214176877185396380, "num_range": 9082589991459931387, "upper_bound": "A9031F39/35D4A066"}, {"distinct_range": 0, "num_eq": 2000000000, "num_range": 6679390017426143194, "upper_bound": "7ED99E47/53D83B5F"}, {"distinct_range": 0, "num_eq": 40, "num_range": 2000000, "upper_bound": "F330F0DE/8663D91C"}, {"distinct_range": 20.25776691271769, "num_eq": 8333683413161511272, "num_range": 30, "upper_bound": "41C0AC17/D906BD2C"}, {"distinct_range": 0, "num_eq": 60000, "num_range": 0, "upper_bound": "8FE77FE7/A5EBDC12"}, {"distinct_range": 0, "num_eq": 3000000000, "num_range": 0, "upper_bound": "6FCDCC80/F76D1531"}], "histo_col_type": "PG_LSN", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 30000}, {"avg_size": 24, "columns": ["\\"col4%q_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 13108, "histo_col_type": "", "name": "__auto__", "null_count": 19406, "row_count": 30000}, {"avg_size": 31, "columns": ["col4_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 22663, "histo_buckets": [{"distinct_range": 0, "num_eq": 50000000000, "num_range": 0, "upper_bound": "-557909844937072136"}, {"distinct_range": 4610265064156021000, "num_eq": 200000, "num_range": 6233824384562303201, "upper_bound": "-7659847884381625915"}], "histo_col_type": "INT8", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 30000}, {"avg_size": 11, "columns": ["col4_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 15523, "histo_col_type": "", "name": "__auto__", "null_count": 16991, "row_count": 30000}]':::JSONB;
    rsg_test.go:720: ALTER TABLE "tab
le_5" INJECT STATISTICS e'[{"avg_size": 13, "columns": ["\\"%pc%pol\\t5_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 82970691375, "histo_buckets": [{"distinct_range": 0, "num_eq": 300, "num_range": 0, "upper_bound": "\\u000e"}, {"distinct_range": 0, "num_eq": 100000, "num_range": 5000000, "upper_bound": "8n"}, {"distinct_range": 0, "num_eq": 1819284330617397464, "num_range": 80000, "upper_bound": "\\u0006\\u001aOgSc_"}, {"distinct_range": 5552.222651027743, "num_eq": 734655193730245714, "num_range": 7000, "upper_bound": "\\u000bHHw\\u0010Fls"}], "histo_col_type": "NAME", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 100000000000}, {"avg_size": 20, "columns": ["\\"c\\"\\"ol5_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 82534256961, "histo_buckets": [{"distinct_range": 0, "num_eq": 5849176931489746032, "num_range": 0, "upper_bound": ""}, {"distinct_range": 300000, "num_eq": 5198912887619801878, "num_range": 300000, "upper_bound": "/Q)\\u0019\\u000b<"}, {"distinct_range": 1000000000, "num_eq": 100, "num_range": 1000000000, "upper_bound": "C[xX&A\\u0006L"}, {"distinct_range": 40, "num_eq": 60000000000, "num_range": 40, "upper_bound": "D\\u001f$\\u0017&/\\u0019\\u0007s"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 100000000000}, {"avg_size": 21, "columns": ["\\"col 5%v_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 68128774871, "histo_buckets": [{"distinct_range": 0, "num_eq": 6422635731287690959, "num_range": 0, "upper_bound": ""}, {"distinct_range": 6959419914340679000, "num_eq": 10000, "num_range": 6959419914340678783, "upper_bound": "<"}, {"distinct_range": 4320035958.268606, "num_eq": 100000, "num_range": 6000000000, "upper_bound": "X"}, {"distinct_range": 30000000000, "num_eq": 8795659030038269644, "num_range": 30000000000, "upper_bound": "%On"}, {"distinct_range": 19.610424650509923, "num_eq": 5091537797586708880, "num_range": 90, "upper_bound": "+z6b"}, {"distinct_range": 0, "num_eq": 6000, "num_range": 2501537819896505355, "upper_bound": "l|u\\u001f"}, {"distinct_range": 0, "num_eq": 8000000000, "num_range": 1000, "upper_bound": "oe-JYJ1\'q"}, {"distinct_range": 3915469917750181000, "num_eq": 900000, "num_range": 6813600572411473950, "upper_bound": "\x7f%\\u001bfov2D}"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 100000000000}]':::JSONB;
    rsg_test.go:720: CREATE TYPE rand_typ_0 AS ENUM ('hasjh', 'd', 'twi', 'pvlutl', 'smzpxp');
    rsg_test.go:720: CREATE TYPE rand_typ_1 AS ENUM ('svdu', 'sn');
    rsg_test.go:720: CREATE TYPE rand_typ_2 AS ENUM ('cqvlzr');
    rsg_test.go:720: CREATE TYPE rand_typ_3 AS ENUM ('idp', 'mdmx', 'u', 'qssf', 'wnr', 'xrz');
    rsg_test.go:720: SET sql_safe_updates = false;;
    rsg_test.go:720: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (9054625319608892284);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (5924588545498469292);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (3299719825395071631);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (995401221845224147);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (1076125818020080009);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (7933405498286966106);
    rsg_test.go:720: ALTER TABLE seed SPLIT AT VALUES (4083943326136135373);
    rsg_test.go:720: ALTER TABLE seed SCATTER;
    rsg_test.go:722: 
    rsg_test.go:723: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith445115098
--- FAIL: TestRandomSyntaxSQLSmith (389.85s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #119441 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot T-sql-foundations branch-release-23.2.1-rc release-blocker]
- #119188 sql/tests: TestRandomSyntaxSQLSmith failed [expected exactly one ordering column for RANGE mode with offset] [A-sql-optimizer C-bug C-test-failure O-robot P-3 T-sql-queries branch-release-23.2]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>



Jira issue: CRDB-36227

--- a/pkg/sql/logictest/testdata/logic_test/tsvector
+++ b/pkg/sql/logictest/testdata/logic_test/tsvector
@@ -457,3 +457,22 @@ SELECT * FROM cte1 GROUP BY cte1.col1;
 ----
 ("",1)
 ("",2)
+
+# Regression test for incorrectly using key-encoding on TSQuery type in the
+# row container infrastructure (we use the row-based windower because json_agg
+# is unsupported as vectorized aggregate function).
+query T rowsort
+WITH
+    cte (col1) AS (VALUES ('foo':::TSQUERY), ('bar':::TSQUERY)),
+    seed AS (SELECT g::INT8 AS _int8, g::STRING::JSONB AS _jsonb FROM generate_series(1, 3) AS g)
+SELECT
+    json_agg(seed._jsonb) OVER (PARTITION BY cte.col1)
+FROM
+    cte, seed;
+----
+[1, 2, 3]
+[1, 2, 3]
+[1, 2, 3]
+[1, 2, 3]
+[1, 2, 3]
+[1, 2, 3]

--- a/pkg/sql/rowcontainer/BUILD.bazel
+++ b/pkg/sql/rowcontainer/BUILD.bazel
@@ -31,6 +31,7 @@ go_library(
         "//pkg/util/bufalloc",
         "//pkg/util/cancelchecker",
         "//pkg/util/encoding",
+        "//pkg/util/errorutil/unimplemented",
         "//pkg/util/hlc",
         "//pkg/util/log",
         "//pkg/util/mon",

--- a/pkg/sql/rowcontainer/disk_row_container.go
+++ b/pkg/sql/rowcontainer/disk_row_container.go
@@ -24,6 +24,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/util/bufalloc"
 	"github.com/cockroachdb/cockroach/pkg/util/encoding"
+	"github.com/cockroachdb/cockroach/pkg/util/errorutil/unimplemented"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/mon"
 	"github.com/cockroachdb/errors"
@@ -97,15 +98,15 @@ var _ DeDupingRowContainer = &DiskRowContainer{}
 //   - e is the underlying store that rows are stored on.
 func MakeDiskRowContainer(
 	diskMonitor *mon.BytesMonitor,
-	types []*types.T,
+	typs []*types.T,
 	ordering colinfo.ColumnOrdering,
 	e diskmap.Factory,
-) DiskRowContainer {
+) (DiskRowContainer, error) {
 	diskMap := e.NewSortedDiskMap()
 	d := DiskRowContainer{
 		diskMap:     diskMap,
 		diskAcc:     diskMonitor.MakeBoundAccount(),
-		types:       types,
+		types:       typs,
 		ordering:    ordering,
 		diskMonitor: diskMonitor,
 		engine:      e,
@@ -140,9 +141,15 @@ func MakeDiskRowContainer(
 	d.encodings = make([]catenumpb.DatumEncoding, len(d.ordering))
 	for i, orderInfo := range ordering {
 		d.encodings[i] = rowenc.EncodingDirToDatumEncoding(orderInfo.Direction)
+		switch t := typs[orderInfo.ColIdx]; t.Family() {
+		case types.TSQueryFamily, types.TSVectorFamily:
+			return DiskRowContainer{}, unimplemented.NewWithIssueDetailf(
+				92165, "", "can't order by column type %s", t.SQLStringForError(),
+			)
+		}
 	}
 
-	return d
+	return d, nil
 }
 
 // DoDeDuplicate causes DiskRowContainer to behave as an implementation of
@@ -326,7 +333,10 @@ func (d *DiskRowContainer) Sort(context.Context) {}
 func (d *DiskRowContainer) Reorder(ctx context.Context, ordering colinfo.ColumnOrdering) error {
 	// We need to create a new DiskRowContainer since its ordering can only be
 	// changed at initialization.
-	newContainer := MakeDiskRowContainer(d.diskMonitor, d.types, ordering, d.engine)
+	newContainer, err := MakeDiskRowContainer(d.diskMonitor, d.types, ordering, d.engine)
+	if err != nil {
+		return err
+	}
 	i := d.NewFinalIterator(ctx)
 	defer i.Close()
 	for i.Rewind(); ; i.Next() {

--- a/pkg/sql/rowcontainer/disk_row_container_test.go
+++ b/pkg/sql/rowcontainer/disk_row_container_test.go
@@ -164,7 +164,7 @@ func TestDiskRowContainer(t *testing.T) {
 				}
 				row := randgen.RandEncDatumRowOfTypes(rng, typs)
 				func() {
-					d := MakeDiskRowContainer(diskMonitor, typs, ordering, tempEngine)
+					d, _ := MakeDiskRowContainer(diskMonitor, typs, ordering, tempEngine)
 					defer d.Close(ctx)
 					if err := d.AddRow(ctx, row); err != nil {
 						t.Fatal(err)
@@ -227,7 +227,7 @@ func TestDiskRowContainer(t *testing.T) {
 			types := randgen.RandSortingTypes(rng, numCols)
 			rows := randgen.RandEncDatumRowsOfTypes(rng, numRows, types)
 			func() {
-				d := MakeDiskRowContainer(diskMonitor, types, ordering, tempEngine)
+				d, _ := MakeDiskRowContainer(diskMonitor, types, ordering, tempEngine)
 				defer d.Close(ctx)
 				for i := 0; i < len(rows); i++ {
 					if err := d.AddRow(ctx, rows[i]); err != nil {
@@ -309,7 +309,7 @@ func TestDiskRowContainer(t *testing.T) {
 		// Use random types and random rows.
 		types := randgen.RandSortingTypes(rng, numCols)
 		numRows, rows := makeUniqueRows(t, &evalCtx, rng, numRows, types, ordering)
-		d := MakeDiskRowContainer(diskMonitor, types, ordering, tempEngine)
+		d, _ := MakeDiskRowContainer(diskMonitor, types, ordering, tempEngine)
 		defer d.Close(ctx)
 		d.DoDeDuplicate()
 		addRowsRepeatedly := func() {
@@ -351,7 +351,7 @@ func TestDiskRowContainer(t *testing.T) {
 		types := randgen.RandSortingTypes(rng, numCols)
 		rows := randgen.RandEncDatumRowsOfTypes(rng, numRows, types)
 		// There are no ordering columns when using the numberedRowIterator.
-		d := MakeDiskRowContainer(diskMonitor, types, nil, tempEngine)
+		d, _ := MakeDiskRowContainer(diskMonitor, types, nil, tempEngine)
 		defer d.Close(ctx)
 		for i := 0; i < numRows; i++ {
 			require.NoError(t, d.AddRow(ctx, rows[i]))
@@ -439,7 +439,7 @@ func TestDiskRowContainerDiskFull(t *testing.T) {
 	)
 	monitor.Start(ctx, nil, mon.NewStandaloneBudget(0 /* capacity */))
 
-	d := MakeDiskRowContainer(
+	d, _ := MakeDiskRowContainer(
 		monitor,
 		[]*types.T{types.Int},
 		colinfo.ColumnOrdering{colinfo.ColumnOrderInfo{ColIdx: 0, Direction: encoding.Ascending}},
@@ -480,7 +480,7 @@ func TestDiskRowContainerFinalIterator(t *testing.T) {
 	diskMonitor.Start(ctx, nil /* pool */, mon.NewStandaloneBudget(math.MaxInt64))
 	defer diskMonitor.Stop(ctx)
 
-	d := MakeDiskRowContainer(diskMonitor, types.OneIntCol, nil /* ordering */, tempEngine)
+	d, _ := MakeDiskRowContainer(diskMonitor, types.OneIntCol, nil /* ordering */, tempEngine)
 	defer d.Close(ctx)
 
 	const numCols = 1
@@ -608,7 +608,7 @@ func TestDiskRowContainerUnsafeReset(t *testing.T) {
 	)
 	monitor.Start(ctx, nil, mon.NewStandaloneBudget(math.MaxInt64))
 
-	d := MakeDiskRowContainer(monitor, types.OneIntCol, nil /* ordering */, tempEngine)
+	d, _ := MakeDiskRowContainer(monitor, types.OneIntCol, nil /* ordering */, tempEngine)
 	defer d.Close(ctx)
 
 	const (

--- a/pkg/sql/rowcontainer/hash_row_container.go
+++ b/pkg/sql/rowcontainer/hash_row_container.go
@@ -115,6 +115,7 @@ func (e *columnEncoder) init(typs []*types.T, keyCols columns, encodeNull bool) 
 // If the row contains any NULLs and encodeNull is false, hasNull is true and
 // no encoding is returned. If encodeNull is true, hasNull is never set.
 func encodeColumnsOfRow(
+	ctx context.Context,
 	da *tree.DatumAlloc,
 	appendTo []byte,
 	row rowenc.EncDatumRow,
@@ -126,12 +127,25 @@ func encodeColumnsOfRow(
 		if row[colIdx].IsNull() && !encodeNull {
 			return nil, true, nil
 		}
-		// Note: we cannot compare VALUE encodings because they contain column IDs
-		// which can vary.
-		// TODO(radu): we should figure out what encoding is readily available and
-		// use that (though it needs to be consistent across all rows). We could add
-		// functionality to compare VALUE encodings ignoring the column ID.
-		appendTo, err = row[colIdx].Encode(colTypes[i], da, catenumpb.DatumEncoding_ASCENDING_KEY, appendTo)
+		ed := row[colIdx]
+		if t := colTypes[i]; t.Family() == types.JsonFamily {
+			// JSON type is special because for historical reasons it uses value
+			// encoding in Fingerprint, yet we do have key encoding available
+			// (and this is what will be used by the disk row container if we
+			// happen to spill to disk).
+			appendTo, err = row[colIdx].Encode(t, da, catenumpb.DatumEncoding_ASCENDING_KEY, appendTo)
+		} else {
+			// Fingerprint internally might decode the EncDatum and will attempt
+			// to update the memory account accordingly. We don't have a
+			// suitable memory account in scope, so we will ensure to lose the
+			// reference to the decoded datum if it wasn't already present, and
+			// then we can pass nil acc argument.
+			hadDecodedDatum := ed.Datum != nil
+			appendTo, err = ed.Fingerprint(ctx, t, da, appendTo, nil /* acc */)
+			if !hadDecodedDatum {
+				ed.Datum = nil
+			}
+		}
 		if err != nil {
 			return appendTo, false, err
 		}
@@ -146,7 +160,7 @@ func (e *columnEncoder) encodeEqualityCols(
 	ctx context.Context, row rowenc.EncDatumRow, eqCols columns,
 ) ([]byte, error) {
 	encoded, hasNull, err := encodeColumnsOfRow(
-		&e.datumAlloc, e.scratch, row, eqCols, e.keyTypes, e.encodeNull,
+		ctx, &e.datumAlloc, e.scratch, row, eqCols, e.keyTypes, e.encodeNull,
 	)
 	if err != nil {
 		return nil, err
@@ -158,6 +172,8 @@ func (e *columnEncoder) encodeEqualityCols(
 	return encoded, nil
 }
 
+var hashKeyEncodingDirection = encoding.Ascending
+
 // storedEqColsToOrdering returns an ordering based on storedEqCols to be used
 // by the row containers (this will result in rows with the same equality
 // columns occurring contiguously in the keyspace).
@@ -166,7 +182,7 @@ func storedEqColsToOrdering(storedEqCols columns) colinfo.ColumnOrdering {
 	for i := range ordering {
 		ordering[i] = colinfo.ColumnOrderInfo{
 			ColIdx:    int(storedEqCols[i]),
-			Direction: encoding.Ascending,
+			Direction: hashKeyEncodingDirection,
 		}
 	}
 	return ordering
@@ -459,7 +475,7 @@ func (i *hashMemRowIterator) computeKey() error {
 		i.curKey, err = i.scratchEncRow[col].Encode(
 			i.container.types[col],
 			&i.container.columnEncoder.datumAlloc,
-			catenumpb.DatumEncoding_ASCENDING_KEY,
+			rowenc.EncodingDirToDatumEncoding(hashKeyEncodingDirection),
 			i.curKey,
 		)
 		if err != nil {
@@ -548,8 +564,9 @@ func (h *HashDiskRowContainer) Init(
 		)
 	}
 
-	h.DiskRowContainer = MakeDiskRowContainer(h.diskMonitor, storedTypes, storedEqColsToOrdering(storedEqCols), h.engine)
-	return nil
+	var err error
+	h.DiskRowContainer, err = MakeDiskRowContainer(h.diskMonitor, storedTypes, storedEqColsToOrdering(storedEqCols), h.engine)
+	return err
 }
 
 // AddRow adds a row to the HashDiskRowContainer. This row is unmarked by

--- a/pkg/sql/rowcontainer/kvstreamer_result_disk_buffer.go
+++ b/pkg/sql/rowcontainer/kvstreamer_result_disk_buffer.go
@@ -66,12 +66,16 @@ func (b *kvStreamerResultDiskBuffer) Serialize(
 	ctx context.Context, r *kvstreamer.Result,
 ) (resultID int, _ error) {
 	if !b.initialized {
-		b.container = MakeDiskRowContainer(
+		var err error
+		b.container, err = MakeDiskRowContainer(
 			b.monitor,
 			inOrderResultsBufferSpillTypeSchema,
 			colinfo.ColumnOrdering{},
 			b.engine,
 		)
+		if err != nil {
+			return 0, err
+		}
 		b.initialized = true
 		b.rowScratch = make(rowenc.EncDatumRow, len(inOrderResultsBufferSpillTypeSchema))
 	}

--- a/pkg/sql/rowcontainer/row_container.go
+++ b/pkg/sql/rowcontainer/row_container.go
@@ -626,7 +626,10 @@ func (f *DiskBackedRowContainer) SpillToDisk(ctx context.Context) error {
 	if f.UsingDisk() {
 		return errors.New("already using disk")
 	}
-	drc := MakeDiskRowContainer(f.diskMonitor, f.mrc.types, f.mrc.ordering, f.engine)
+	drc, err := MakeDiskRowContainer(f.diskMonitor, f.mrc.types, f.mrc.ordering, f.engine)
+	if err != nil {
+		return err
+	}
 	f.src = &drc
 	f.drc = &drc
 	if f.deDuplicate {

